### PR TITLE
Fixes issue with navigationGuard by starting oktaAuth service for login redirect

### LINKED ISSUE #72: App stuck on initial load after token times out
## I'm submitting a:

- [X] Bug report  
- [ ] Feature request
- [ ] Other (Describe below)

## Current behavior

I've noticed that whenever I navigate to my app after ~30 minutes, it will perform the authorization and token calls when entering but doesn't seem to trigger some sort or refresh. The URL is accurate but the page component is not rendered. If I refresh the page, everything works fine. All of my pages have `requiredAuth: true`. Note: this doesn't have to do with the redirectUri as the app loads fine when it goes to Okta for login. It seems like it has to do with using cached information and performing some sort of auth check.

## Expected behavior

When first entering my site it should authorize if need be and render the page component.

## Minimal reproduction of the problem with instructions

Here is my OktaAuth setup:
```js
export default new OktaAuth({
  issuer: window.env.OKTA_ISSUER,
  clientId: window.env.OKTA_CLIENT_ID,
  redirectUri: window.location.origin + '/implicit/callback',
  scopes: ['openid', 'profile', 'email', 'groups'],
  pkce: true,
});
```

This is how my routes are set up:
```js
{
  path: '/',
  name: 'Home',
  component: () => import(/* webpackChunkName: "home" */ '../components/Home.vue'),
  meta: {
    requiresAuth: true,
  },
},
```

and I use the navigation guard as well:
```js
router.beforeEach(navigationGuard);
```

## Environment

- Package version: 5.0.1
- Vue version: 3.2.1
- Browser: Chrome 
- OS: MacoS
- Node version (`node -v`): Not running locally
- Other: @okta/okta-auth-js 5.3.0

### PR DESCRIPTION
**Pull request by denysoblohin-okta**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [guidelines](/okta/okta-vue/blob/master/CONTRIBUTING.md#commit)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Adding Tests
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation changes
- [ ] Other... Please describe:

## What is the current behavior?

After expiring of tokens when user tries to navigate to protected page, `navigationGuard` will be called. 
`_oktaAuth.isAuthenticated()` will return false, but `_oktaAuth.authStateManager.getAuthState()` will contain `isAuthenticated: true`, so `guardSecureRoute` will do nothing.

Issue Number: OKTA-420066
Resolves #72 

## What is the new behavior?
Need to start oktaAuth service after `handleLoginRedirect()` call in `LoginCallback` component. This way `isAuthenticated` will be correct.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information
Using okta-auth-js 5.4.3+ will fix issue (see [comment](https://github.com/okta/okta-vue/issues/72#issuecomment-899705911)) because of using default `autoRenew: true` in `isAuthenticated()` call which will remove expired tokens.

Updated `okta-auth-js` to `^5.8.0`

### CODE REVIEW

**Comment by shuowu** [comment]
I think the issue comes from https://github.com/okta/okta-vue/blob/master/src/okta-vue.ts#L120, with the latest version of okta-auth-js, we should just call the oktaAuth.start(), then handleLoginRedirect should be able to update the tokens when it's finished. 

This logic has been updated in okta-react and okta-angular, but by some reason this repo has been missed.

https://github.com/okta/okta-angular/blob/master/src/okta/okta.module.ts#L77
https://github.com/okta/okta-react/blob/master/src/Security.tsx#L72

**Comment by denysoblohin-okta** [comment]
Changed to always start service

File: src/okta-vue.ts [hunk 1]
```diff
@@ -53,7 +53,9 @@ export const navigationGuard = async (to: RouteLocationNormalized) => {
     const isAuthenticated = await _oktaAuth.isAuthenticated()
     if (!isAuthenticated) {
       const authState = _oktaAuth.authStateManager.getAuthState()
-      await guardSecureRoute(authState)
+      if (authState) {
```
Comment (shuowu) [inline src/okta-vue.ts:56]: nit: how about handle it in `guardSecureRoute`, we can just do nothing when `authState` is null (app initial state and tokens has not been loaded from storage)

**Review by shuowu — COMMENTED** [review]
This one looks like a patch fix, please base it against the proper version branch and add changelog.

File: src/okta-vue.ts (hunk 1 above)
Comment (denysoblohin-okta) [reply src/okta-vue.ts]: Done

**Comment by denysoblohin-okta** [comment]
@shuo Rebased onto `5.0`, added changelog (5.0.2)

**Review by shuowu — APPROVED** [review]

**Pull request closed without merging**